client.py

```diff
@@ -82,4 +84,29 @@ async def roll(ctx: SlashContext, num_sides: int = 6, num_dice: int = 2):
   await ctx.send(content=f"Rolled {num_dice} {num_sides}-sided dice. Output: {rolls}, Total: {sum(rolls)}")
 
 
+@slash.slash(
+    name="save",
+    guild_ids=GUILD_IDS,
+    description="Update a key/value",
+    options=[
+        create_option(
+            name="key",
+            description="key",
+            option_type=SlashCommandOptionType.STRING,
+            required=True,
+        ),
+        create_option(
+            name="value",
+            description="value",
+            option_type=SlashCommandOptionType.STRING,
+            required=True,
+        ),
+    ])
+async def save(ctx: SlashContext, key, value):
+  ds.data[key] = value

```

Comment:
since `data` is a function. Why does it not need parentheses like `ds.data()[key]`?

Reply:
python has this feature called “properties” which is why it’s annotated with the `@property` decorator. It basically creates a syntax sugar so that when you use the object it automatically calls the getter property.

e.g. if i did
```
abc = ds.data
```
That’s actually the same thing as 
```
abc = ds.data() # without property decorator
```

Reply:
You can also have setter properties, so that when you do an assignment like
```
ds.data = xyz
```

that translates to
```
ds.set_data(xyz)
```